blaze/mahimahi/server/server.py

```diff
@@ -29,39 +29,46 @@ def prepend_javascript_snippet(input_string: str):
-function getCriticalRequests(){importantRequests=[]
-importantRequests=imagesInViewPort.map((url)=>(url));urlRequestors.forEach((k)=>{if(imagesInViewPort.includes(k.url)){importantRequests=importantRequests.concat(k.initiator)}})
-return importantRequests}
-function findAndPrintImagesInViewport(ele){ele.querySelectorAll('*').forEach(function(node){if(node.tagName=="IMG"){let url=null;console.log("alohomoradebug checking node "+JSON.stringify(node))
-if(isElementInViewport(node)){console.log("alohomoradebug: element is in viewport")
-if(typeof node.href!='undefined'){url=node.href}
-if(typeof node.src!='undefined'){url=node.src}
-if(url!=null){imagesInViewPort.push(url)}}else{console.log("alohomoradebug: element not in viewport")}}});let answer=getCriticalRequests()
-console.log("alohomoradebug: this is the answer")
-console.log(JSON.stringify({'alohomora_output':answer}))
-console.log("alohomoradebug: that was the the answer")}
-window.addEventListener('load',(event)=>{console.log("alohomoradebug: extracting critical elements")
-document.querySelectorAll("iframe").forEach((ele)=>{var iframeID=ele.id;iframeElement=document.getElementById(iframeID)
-if(isAnyPartOfElementInViewport(iframeElement)){var innerDoc=(iframeElement.contentDocument)?iframeElement.contentDocument:iframeElement.contentWindow.document;findAndPrintImagesInViewport(innerDoc)}});findAndPrintImagesInViewport(document)})
-    """
-    if soup.html is not None:
-        soup.html.insert(0, critical_catcher)
-        soup.html.insert(0, stack_trace_dependency)
+    if soup.html is None:
+        return str(soup)
+    dir_name = "/opt/blaze/blaze/mahimahi/server/injected-javascript"
+    for file_name in ["find-in-viewport.js", "interceptor.js"]:
+        curr_file = os.path.join(dir_name, file_name)
+        with open(curr_file) as f:
+            file_contents = f.read()
+            script_tag = soup.new_tag("script")
+            script_tag["type"] = "application/javascript"
+            script_tag.string = file_contents
+            soup.html.insert(0, script_tag)
     return str(soup)
 
 
+def inject_extract_critical_requests_javascript(file):
+    """
+    tries to inject the two js files in the given file.
+    throws exception if an error happens
+    returns body as string after injection
+    """
+    uncompressed_body = file.body
+    gzipped_file = False
+    if (
+        file.headers is not None
+        and file.headers.get("content-encoding") is not None
+        and "gzip" in file.headers.get("content-encoding")
+    ):
+        gzipped_file = True
+        uncompressed_body = gzip.GzipFile(fileobj=BytesIO(uncompressed_body)).read()
+    uncompressed_body = prepend_javascript_snippet(uncompressed_body)
+    if gzipped_file:
+        out = BytesIO()
+        with gzip.GzipFile(fileobj=out, mode="wb") as f:
+            f.write(uncompressed_body.encode())
+        file.body = out.getvalue()
+    else:
```

Comment:
do we really need this else statement

Reply:
this is essentially decompressing the file.body and saving it to back to file.body.. 
we could save the output in a different variable instead of overwriting file.body maybe?

Reply:
in the case of else, the file body is being written to what was already in it